utilities.py

```diff
@@ -0,0 +1,202 @@
+
+    Returns
+    -------
+    datetime.date
+        The date of the month_id, using first day of the month format.
+    """
+    df = pd.DataFrame()
+    df["year"] = views_month_id_to_year(month_id)
+    df["month"] = views_month_id_to_month(month_id)
+    df["day"] = 1
+    return pd.to_datetime(df, format = "%Y%M")
+
+def migrate_submission_from_old(submission: str|os.PathLike) -> None:
+    """Helper function to migrate old submission folder structure to new structure based on Apache Hive.
+
+    Parameters
+    ----------
+    submission : str | os.PathLike
+        Folder following the old submission_template setup
+    """
+
+    submission = Path(submission)
+    eval_data = itertools.chain(submission.glob(f'**/cm/*/eval/*.parquet'),
+                                submission.glob(f'**/pgm/*/eval/*.parquet'))
+        
+    def folder_rename(d):
+        new_name = d.parent / f'window=Y{d.name.split("_")[-1]}'
+        d.rename(new_name)
+
+    # Rename window folders in Apache Hive format
+    window_folders = itertools.chain(submission.glob(f'cm/test_window_*'), submission.glob(f'pgm/test_window_*'))
+    [folder_rename(d) for d in window_folders]
+    # Delete evaluation files (must be re-estimated)
+    [f.unlink() for f in eval_data]        
+    [f.unlink() for f in submission.glob(f'eval*.parquet')]
+    
+    # Cleanup old folders (only deletes if empty)
+    [d.rmdir() for d in submission.glob(f'**/cm/*/eval/')]
+    [d.rmdir() for d in submission.glob(f'**/pgm/*/eval/')]
+
+def list_submissions(submissions_folder: str|os.PathLike) -> list[os.PathLike]:
+    """Creates a list of paths to folders inside the submissions_folder.
+
+    Parameters
+    ----------
+    submissions_folder : str | os.PathLike
+        Path to a folder only containing folders structured like a submission_template
+
+    Returns
+    -------
+    list[os.PathLike]
+        List of paths to submissions.
+    """
+    submissions_folder = Path(submissions_folder)
+    return [submission for submission in submissions_folder.iterdir() if submission.is_dir() and not submission.stem == "__MACOSX"]
+    
+def read_parquet(data_path: str|os.PathLike, filters = None) -> pd.DataFrame:
+    """This function does not need to be used directly, it is mostly to document how to read folders with parquet files or single parquet files with a filter.
+    
+    Notes
+    -----
+
+    See https://arrow.apache.org/docs/python/compute.html#filtering-by-expressions for filter examples
+    
+    Examples
+    --------
+    >>> import pyarrow.compute as pac
+    >>> import pyarrow.parquet as pq
+    >>> filter = pac.field("year") >= 2017
+    >>> df = pq.ParquetDataset(path_to_folder_with_apache_hive_structured_parquet_files, filters = filter).read().to_pandas()
+
+    """
+
+    table = pq.ParquetDataset(data_path, filters = filters)
+    return table.read().to_pandas()
+
+def data_in_target(submission: str|os.PathLike, target: TargetType) -> bool:
```

Comment:
I may suggest that this function might be named to is_there_parquet_in_folder for improved clarity and focus.